### PR TITLE
Added documentation for viewer message API.

### PR DESCRIPTION
**Pull request by felixhageloh**
- added section to landing page
- added the new docs
- tweaked the JS API docs to match
- tweaked code block styles

screen shot 2015-10-29 at 10 08 22

### CODE REVIEW

**Comment by Kureev** [comment]
Wow, great job! 
image

File: source/viewer-message-api.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+      // do something with eventData
+      break;
+    case 'event type B'
+      // do something with eventData
+      break;
+  }
+}
+```
+
+Once you've parsed the message to get the event data, you can react on the event type in your code. Currently only one event type is supported:
```
Comment (unimatrixZxero) [inline source/viewer-message-api.md]: "can react **to**..." sounds more natural to me. #my_2¢

File: source/viewer-message-api.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  data.state.productId // e.g. 2431
+}
+```
+
+This event is triggered whenever a user does one of two things:
+
+- navigates to a page/spread
+- clicks on a product hotspot
+
+Thus any action in the Viewer that reveals content which has its own url.
```
Comment (unimatrixZxero) [inline source/viewer-message-api.md]: URL

File: source/viewer-message-api.md (hunk 1 above)
Comment (felixhageloh) [inline source/viewer-message-api.md]: to me as well - not sure what I was thinking :confounded:

File: source/viewer-message-api.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+This API can be used to listen to events in the Viewer, for example:
+
+- the user navigated to a page/spread
+- the user opened a product
+
+
+# Getting Started
+
+The message API uses HTML5s [window.postMessage](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) to send messages. You can access messages sent by the Viewer in two different ways:
```
Comment (unimatrixZxero) [inline source/viewer-message-api.md]: "...uses HTML5**'**s ..."

File: source/viewer-message-api.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+### Using the Code Injection Tool
+
+```javascript
+window.addEventListener('message', function(message) {
+  // do something with message
+})
+```
+
+The code injection tool in the [Publitas CMS](revolution.publitas.com) lets you add custom HTML tags into the header of a publication. Use a `<script>` tag to include custom javacsript code.
+The viewer will post messages to its window, so we can listen to to events using `window.addEventListener`
```
Comment (unimatrixZxero) [inline source/viewer-message-api.md]: remove double `to`

**Comment by unimatrixZxero** [comment]
:+1:

**Pull request merged**